action/main.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/go-github/v37/github"
+	"github.com/sethvargo/go-githubactions"
+)
+
+func main() {
+	client := initializeGithubClient()
+
+	client.ValidateTargetWorkflowExists(context.Background())
+
+	checkRun := client.CreateCheck(context.Background())
+
+	client.DispatchWorkflow(context.Background(), checkRun)
+
+	waitForCheckCompletion(client, checkRun)
+}
+
+func initializeGithubClient() *GitHubClient {
+	githubVars, err := parseGithubVars()
+	if err != nil {
+		githubactions.Fatalf("%v", err.Error())
+	}
+
+	inputs, err := parseInputs()
+	if err != nil {
+		githubactions.Fatalf("%v", err.Error())
+	}
+
+	client := NewGitHubClient(githubVars, inputs)
+
+	return client
```

Comment:
```suggestion
	return NewGitHubClient(githubVars, inputs)
```

Can tighten this up a bit